apps/src/sites/studio/pages/components/StageCard.jsx

```diff
@@ -0,0 +1,166 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { connect } from 'react-redux';
+import { borderRadius, levelTokenMargin } from './constants';
+import OrderControls from './OrderControls';
+import LevelToken from './LevelToken';
+
+const styles = {
+  checkbox: {
+    margin: '0 0 0 7px'
+  },
+  stageCard: {
+    fontSize: 18,
+    background: 'white',
+    border: '1px solid #ccc',
+    borderRadius: borderRadius,
+    padding: 20,
+    margin: 10
+  },
+  stageCardHeader: {
+    color: '#5b6770',
+    marginBottom: 15
+  },
+  stageLockable: {
+    fontSize: 13,
+    marginTop: 3
+  },
+  addLevel: {
+    fontSize: 14,
+    background: '#eee',
+    border: '1px solid #ddd',
+    boxShadow: 'inset 0 1px 0 0 rgba(255, 255, 255, 0.8)',
+    margin: '5px 0'
+  }
+};
+
+const StageCard = React.createClass({
+  propTypes: {
+    reorderLevel: React.PropTypes.func.isRequired,
+    addLevel: React.PropTypes.func.isRequired,
+    stagesCount: React.PropTypes.number.isRequired,
+    stage: React.PropTypes.object.isRequired
+  },
+
+  getInitialState() {
+    return {currentPositions: []};
+  },
+
+  metrics(token) {
+    return ReactDOM.findDOMNode(this.refs[`levelToken${token}`]).getBoundingClientRect();
```

Comment:
@Bjvanminnen Have you read through http://stackoverflow.com/questions/37468913/why-ref-string-is-legacy? A lot of stuff there (like static analysis type checkers) do not apply to us at the moment, but the way this breaks potential VM optimizations might be interesting for us depending on the specific scenario.

Anyway, I don't have strong opinions about string vs callback refs in our code base at large, though whenever I see string concatenation involved in object property lookups, I raise my eyebrows ever so slightly.